postem/postem-app/src/main/java/org/sakaiproject/postem/controller/MainController.java

```diff
@@ -0,0 +1,114 @@
+@Slf4j
+@Controller
+public class MainController {
+
+    @Autowired
+    private PostemSakaiService postemSakaiService;
+	
+    @Autowired
+    private SessionManager sessionManager;
+    
+    @Autowired
+    private PreferencesService preferencesService;
+
+    @GetMapping(value = {"/", "/index"})
+    public String showIndex(Model model, HttpServletRequest request, HttpServletResponse response) {
+        log.debug("showIndex");
+        
+        String userId = sessionManager.getCurrentSessionUserId();
+        final Locale locale = StringUtils.isNotBlank(userId) ? preferencesService.getLocale(userId) : Locale.getDefault();
+        LocaleResolver localeResolver = RequestContextUtils.getLocaleResolver(request);
+        localeResolver.setLocale(request, response, locale);
+		
+		List<Gradebook> gradebooksList = postemSakaiService.getGradebooks(Gradebook.SORT_BY_TITLE, true);
+		
+		model.addAttribute("gradebooksList", gradebooksList);
+		model.addAttribute("sortedByTitle", PostemToolConstants.POSTEM_TRUE_CONSTANT);
+		model.addAttribute("ascendingTitle", PostemToolConstants.POSTEM_TRUE_CONSTANT);
+		model.addAttribute("ascendingCreator", PostemToolConstants.POSTEM_FALSE_CONSTANT);
+		model.addAttribute("ascendingModifiedBy", PostemToolConstants.POSTEM_FALSE_CONSTANT);
+		model.addAttribute("ascendingLastMod", PostemToolConstants.POSTEM_FALSE_CONSTANT);
+		model.addAttribute("ascendingReleased", PostemToolConstants.POSTEM_FALSE_CONSTANT);	
+		
+		String visible = Boolean.toString(postemSakaiService.checkAccess());
+		model.addAttribute("visible", visible);	
+		
+		ToolSession toolSession = sessionManager.getCurrentToolSession();
+		toolSession.setAttribute("currentGradebook", null);
+
+        return PostemToolConstants.INDEX_TEMPLATE;
+    }
+    
+    @GetMapping(value = {"/add"})
+    public String addItem(@ModelAttribute("gradebookForm") GradebookForm gradebookForm, Model model) {
+        log.debug("addItem");
+
+		ToolSession toolSession = sessionManager.getCurrentToolSession();
+		Gradebook currentGradebook = (Gradebook) toolSession.getAttribute("currentGradebook");
+		
+		if (Objects.nonNull(currentGradebook )) {
+		    String[] parts = currentGradebook.getFileReference().split("/");
+		    String partFileReference = parts[parts.length-1];
+			gradebookForm.setFileReference(partFileReference);			
+			gradebookForm.setTitle(currentGradebook.getTitle());
+			gradebookForm.setReleased(currentGradebook.getReleased());
+		}
+		
+		String visible = Boolean.toString(postemSakaiService.checkAccess());
+		model.addAttribute("visible", visible);	
+		
+  		model.addAttribute("gradebookForm", gradebookForm);
+  		model.addAttribute("fileReference", gradebookForm.getFileReference());
+  		String uploadMax = ServerConfigurationService.getString(PostemToolConstants.SAK_PROP_MAX_UPLOAD_FILE_SIZE);
+  		if (Objects.isNull(uploadMax) || uploadMax.isEmpty()) {
+  			uploadMax = "20"; //default MB
+  		}
+  		model.addAttribute("uploadMax", uploadMax);
```

Comment:
model.addAttribute("uploadMax", serverConfigurationService.getString(PostemToolConstants.SAK_PROP_MAX_UPLOAD_FILE_SIZE, 20));

You need to inject serverConfigurationService, not use the static cover version.

Reply:
Done.